cpp/src/decisiontree/decisiontree_impl.h

```diff
@@ -25,7 +25,7 @@
 #include <map>
 #include <numeric>
 #include <vector>
-#include "levelalgo/metric_def.h"
```

Comment:
metric_def.h was moved to metric_def.cuh as there were 2 Functors that actually used device methods.
However, I see this file didnt use anything from this header but rather `memory.h`. I have changed this.

Reply:
Makes sense. Thanks